modules/core/src/main/java/org/apache/ignite/internal/visor/cache/index/IndexForceRebuildTask.java

```diff
@@ -68,111 +65,60 @@ protected IndexForceRebuildJob(@Nullable IndexForceRebuildTaskArg arg, boolean d
             throws IgniteException
         {
             //Either cacheNames or cacheGrps must be specified
-            assert (arg.cacheNames() == null) != (arg.cacheGrps() == null) :
-                "Either cacheNames or cacheGroups must be specified";
-
-            // Collect info about indexes being rebuilt.
-            Set<IndexRebuildStatusInfoContainer> rebuildIdxCaches =
-                ignite.context().cache().publicCaches()
-                    .stream()
-                    .filter(c -> !c.indexReadyFuture().isDone())
-                    .map(this::fromIgniteCache)
-                    .collect(Collectors.toSet());
-
-            Set<String> rebuildIdxCachesNames = rebuildIdxCaches.stream()
-                .map(IndexRebuildStatusInfoContainer::cacheName)
-                .collect(Collectors.toSet());
+            assert (arg.cacheNames() == null) ^ (arg.cacheGrps() == null) :
+                "Either cacheNames or cacheGroups must be specified.";
 
-            if (arg.cacheNames() != null)
-                return rebuildByCacheNames(arg.cacheNames(), rebuildIdxCaches, rebuildIdxCachesNames);
-            else if (arg.cacheGrps() != null)
-                return rebuildByGroupNames(arg.cacheGrps(), rebuildIdxCaches, rebuildIdxCachesNames);
-            else {
-                assert false : "Neither cache names nor cache groups specified";
+            if (arg.cacheNames() == null && arg.cacheGrps() == null) {
```

Comment:
This condition will not work because there is a XOR above: https://en.wikipedia.org/wiki/Exclusive_or#Truth_table

Reply:
This is should work without -ea launching option.